spyderlib/widgets/sourcecode/codeeditor.py

```diff
@@ -519,13 +530,74 @@ def create_shortcuts(self):
+                                    name='Previous line', parent=self)
+        next_line = create_shortcut(cb_maker('Down'), context='Editor',
+                                    name='Next line', parent=self)
+
+        prev_char = create_shortcut(cb_maker('Left'), context='Editor',
+                                    name='Previous char', parent=self)
+        next_char = create_shortcut(cb_maker('Right'), context='Editor',
+                                    name='Next char', parent=self)
+
+        prev_word = create_shortcut(cb_maker('StartOfWord'), context='Editor',
+                                    name='Previous word', parent=self)
+        next_word = create_shortcut(cb_maker('EndOfWord'), context='Editor',
+                                    name='Next word', parent=self)
+
+        kill_line_end = create_shortcut(self.kill_line_end, context='Editor',
+                                        name='Kill to line end', parent=self)
+        kill_line_start = create_shortcut(self.kill_line_start,
+                                          context='Editor',
+                                          name='Kill to line start',
+                                          parent=self)
+
+        yank = create_shortcut(self._kill_ring.yank, context='Editor',
+                               name='Yank', parent=self)
+        kill_ring_rotate = create_shortcut(self._kill_ring.rotate,
+                                           context='Editor',
+                                           name='Rotate kill ring',
+                                           parent=self)
+
+        kill_prev_word = create_shortcut(self.kill_prev_word, context='Editor',
+                                         name='Kill previous word',
+                                         parent=self)
+        kill_next_word = create_shortcut(self.kill_next_word, context='Editor',
+                                         name='Kill next word', parent=self)
+
+        start_doc = create_shortcut(cb_maker('Start'), context='Editor',
+                                    name='Start of Document', parent=self)
+
+        end_doc = create_shortcut(cb_maker('End'), context='Editor',
+                                  name='End of document', parent=self)
+
         # Fixed shortcuts
         new_shortcut(SHORTCUT_INLINE, self, lambda: self.enter_array_inline())
         new_shortcut(SHORTCUT_TABLE, self, lambda: self.enter_array_table())
 
         return [codecomp, duplicate_line, copyline, deleteline, movelineup,
                 movelinedown, gotodef, toggle_comment, blockcomment,
-                unblockcomment]
+                unblockcomment, line_start, line_end, prev_line, next_line,
+                prev_char, next_char, prev_word, next_word, kill_line_end,
+                kill_line_start, yank, kill_ring_rotate, kill_prev_word,
+                kill_next_word, start_doc, end_doc]
+
+    def cursor_move_event(self, move_type):

```

Comment:
Could you move this method before `cb_maker`? It seems it's only needed there, right?